kcbq-connector/src/integration-test/java/com/wepay/kafka/connect/bigquery/it/BigQueryConnectorIntegrationTest.java

```diff
@@ -176,8 +173,15 @@ private Object convertField(Field fieldSchema, FieldValue field) {
     List<Object> result = new ArrayList<>();
     assert (rowSchema.size() == row.size());
 
+    for (int i=0; i < rowSchema.size(); i++) {
+      if (rowSchema.get(i).getName().equals("row")) {
```

Comment:
The` schema unionization `logic creates fields in the BigQuery table which might not follow the same order as the fields in the `schema.json` file. The integration tests compare the actual and expected rows based on `testRow.get(0`) which is supposed to be the `“row”` field. But since the fields would now be shuffled, I tried to add the `“row”` field values first in the `testRow` list to make sure that it is the first value. Then in the subsequent loop skipped the "row" field values, so that these are not entered twice in the list. But I would want to know if there's a better way.

Reply:
Not sure how generalized indexOf() can be in java, can we use this to make this cleaner?

Edit: looks like we can't specify a comparator for it actually